From 46110b055eaaa0c1f815ff876da4713499c17bc8 Mon Sep 17 00:00:00 2001
From: Max <max@privatevoid.net>
Date: Fri, 17 Jun 2022 22:17:08 +0200
Subject: [PATCH 4/4] Use more NAT traversal features

---
 p2p/node.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/p2p/node.go b/p2p/node.go
index 2f86317..0c9a250 100644
--- a/p2p/node.go
+++ b/p2p/node.go
@@ -49,6 +49,9 @@ func CreateNode(ctx context.Context, inputKey string, port int, handler network.
 		libp2p.NATPortMap(),
 		libp2p.DefaultMuxers,
 		libp2p.Transport(tcp.NewTCPTransport),
+		libp2p.EnableHolePunching(),
+		libp2p.EnableRelayService(),
+		libp2p.EnableNATService(),
 		libp2p.FallbackDefaults,
 	)
 	if err != nil {
-- 
2.36.0

