diff --git a/vendor/github.com/ipfs/go-filestore/fsrefstore.go b/vendor/github.com/ipfs/go-filestore/fsrefstore.go
index 9eb2b43..43e336c 100644
--- a/vendor/github.com/ipfs/go-filestore/fsrefstore.go
+++ b/vendor/github.com/ipfs/go-filestore/fsrefstore.go
@@ -291,9 +291,6 @@ func (f *FileManager) putTo(ctx context.Context, b *posinfo.FilestoreNode, to pu
 			return ErrFilestoreNotEnabled
 		}
 		//lint:ignore SA1019 // ignore staticcheck
-		if !filepath.HasPrefix(b.PosInfo.FullPath, f.root) {
-			return fmt.Errorf("cannot add filestore references outside ipfs root (%s)", f.root)
-		}
 
 		p, err := filepath.Rel(f.root, b.PosInfo.FullPath)
 		if err != nil {
