diff --git a/server/database.js b/server/database.js
index 575827f3..8e288095 100644
--- a/server/database.js
+++ b/server/database.js
@@ -87,20 +87,20 @@ class Database {
      */
     static init(args) {
         // Data Directory (must be end with "/")
-        Database.dataDir = process.env.DATA_DIR || args["data-dir"] || "./data/";
+        Database.dataDir = process.env.DATA_DIR || args["data-dir"] || "./data";
 
         // Plugin feature is working only if the dataDir = "./data";
-        if (Database.dataDir !== "./data/") {
+        if (Database.dataDir !== "./data") {
             log.warn("PLUGIN", "Warning: In order to enable plugin feature, you need to use the default data directory: ./data/");
             PluginsManager.disable = true;
         }
 
-        Database.path = Database.dataDir + "kuma.db";
+        Database.path = Database.dataDir + "/kuma.db";
         if (! fs.existsSync(Database.dataDir)) {
             fs.mkdirSync(Database.dataDir, { recursive: true });
         }
 
-        Database.uploadDir = Database.dataDir + "upload/";
+        Database.uploadDir = Database.dataDir + "/upload/";
 
         if (! fs.existsSync(Database.uploadDir)) {
             fs.mkdirSync(Database.uploadDir, { recursive: true });
@@ -455,7 +455,7 @@ class Database {
     static backup(version) {
         if (! this.backupPath) {
             log.info("db", "Backing up the database");
-            this.backupPath = this.dataDir + "kuma.db.bak" + version;
+            this.backupPath = this.dataDir + "/kuma.db.bak" + version;
             fs.copyFileSync(Database.path, this.backupPath);
 
             const shmPath = Database.path + "-shm";
