--- a/core/corehttp/webui.go	2020-05-09 04:58:18.000000000 +0200
+++ b/core/corehttp/webui.go	2020-07-15 00:22:00.439862559 +0200
@@ -1,7 +1,7 @@
 package corehttp
 
 // TODO: move to IPNS
-const WebUIPath = "/ipfs/bafybeihcyruaeza7uyjd6ugicbcrqumejf6uf353e5etdkhotqffwtguva" // v2.13.0
+const WebUIPath = "/ipns/webui.ipfs.privatevoid.net"
 
 // this is a list of all past webUI paths.
 var WebUIPaths = []string{
