diff --git a/unix_integration/common/src/unix_config.rs b/unix_integration/common/src/unix_config.rs
index 34d8f950f..36773c089 100644
--- a/unix_integration/common/src/unix_config.rs
+++ b/unix_integration/common/src/unix_config.rs
@@ -363,12 +363,49 @@ impl UnixdConfig {
             })
             .collect();
 
+        let service_account_token_path_env = match env::var("KANIDM_SERVICE_ACCOUNT_TOKEN_PATH")
+        {
+            Ok(val) => val.into(),
+            Err(_) => DEFAULT_KANIDM_SERVICE_ACCOUNT_TOKEN_PATH.into(),
+        };
+
+        let service_account_token_path: PathBuf = service_account_token_path_env;
+
+        let service_account_token = if service_account_token_path.exists() {
+            let token_string = read_to_string(&service_account_token_path).map_err(|err| {
+                error!(
+                    ?err,
+                    "Unable to open and read service account token file '{}'",
+                    service_account_token_path.display()
+                );
+                UnixIntegrationError
+            })?;
+
+            let token_string =
+                token_string
+                    .lines()
+                    .next()
+                    .map(String::from)
+                    .ok_or_else(|| {
+                        error!(
+                            "Service account token file '{}' does not contain an api token",
+                            service_account_token_path.display()
+                        );
+                        UnixIntegrationError
+                    })?;
+
+            Some(token_string)
+        } else {
+            // The file does not exist, there is no token to use.
+            None
+        };
+
         let kanidm_config = Some(KanidmConfig {
             conn_timeout: config.conn_timeout.unwrap_or(DEFAULT_CONN_TIMEOUT),
             request_timeout: config.request_timeout.unwrap_or(DEFAULT_CONN_TIMEOUT * 2),
             pam_allowed_login_groups: config.pam_allowed_login_groups.unwrap_or_default(),
             map_group,
-            service_account_token: None,
+            service_account_token,
         });
 
         // Now map the values into our config.
